### PR TITLE
LUN-23: Setup fastlane and build ios

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

**Comment by SteffiStuffel** [comment]
@maxammann 

Maybe also have a look at lunes-credentials.

File: .circleci/src/commands/restore_ruby_cache.yml [hunk 1]
```diff
@@ -5,12 +5,14 @@ parameters:
     default: android
 steps:
   - restore_cache:
-      key: 1-gems-{{ checksum "<< parameters.directory >>/Gemfile.lock" }}
+      keys:
+        - 3-gems-{{ arch }}-{{ checksum "<< parameters.directory >>/Gemfile.lock" }}
+        - 3-gems-{{ arch }}-
   - run:
       name: Install Fastlane
-      command: bundle check || bundle install --path vendor/bundle
```
Comment (klinzo) [inline .circleci/src/commands/restore_ruby_cache.yml:11]: Why did you remove the path argument?

File: ios/fastlane/Appfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
-# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
-# apple_id("[[APPLE_ID]]") # Your Apple email address
-
-
 # For more information about the Appfile, see:
 #     https://docs.fastlane.tools/advanced/#appfile
+
+
+# apple_id "app-team@integreat-app.de"
+# team_id "7272KE28TJ" # Developer Portal Team ID
```
Comment (klinzo) [inline ios/fastlane/Appfile:6]:
```suggestion
```

File: ios/Lunes.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1048,6 +1055,7 @@
 					"\"$(inherited)\"",
 				);
 				MTL_ENABLE_DEBUG_INFO = NO;
+				PROVISIONING_PROFILE_SPECIFIER = None;
```
Comment (klinzo) [inline ios/Lunes.xcodeproj/project.pbxproj:1058]: Is this correct? Seems weird for release builds

File: .circleci/src/jobs/deliver_browser_stack.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+docker:
+  - image: circleci/android:api-29-node
+environment:
+  FASTLANE_SKIP_UPDATE_CHECK: true
+shell: /bin/bash -eo pipefail
+steps:
+  - add_ssh_keys: # Needed for credentials repo
+      fingerprints:
+        - 5b:28:97:36:8b:31:08:ae:5d:ad:19:8a:7d:6f:57:8a
```
Comment (klinzo) [inline .circleci/src/jobs/deliver_browser_stack.yml:9]: Does it contain the credentials of browserstack as well?

File: ios/fastlane/Fastfile [hunk 5]
```diff
@@ -16,8 +16,60 @@
...
+    app_store_connect_api_key(
+        key_id: ENV['APP_STORE_CONNECT_API_KEY_ID'],
+        issuer_id: ENV['APP_STORE_CONNECT_API_ISSUER_ID'],
+        key_content: Base64.decode64(ENV['APP_STORE_CONNECT_API_KEY_CONTENT'])
+    )
   end
+
+  desc "Create a release build"
+  lane :build do |options|
+    setup_circle_ci
```
Comment (klinzo) [inline ios/fastlane/Fastfile:34]: What does this do?

File: ios/fastlane/Fastfile [hunk 6]
```diff
@@ -16,8 +16,60 @@
...
+    version_code = options[:version_code]
+    version_name = options[:version_name]
```
Comment (klinzo) [inline ios/fastlane/Fastfile:36]: I think it would be good to add documentation that this parameters are needed, e.g.
```
  # The following parameters have to be passed:
  # version_code: The version code of the app
  # version_name: The version name of the app
```

**Review by klinzo — APPROVED** [review]
Looking fine :)

File: .circleci/src/commands/restore_ruby_cache.yml (hunk 1 above)
Comment (SteffiStuffel) [reply .circleci/src/commands/restore_ruby_cache.yml:11]: Not necessary. In Integreat it is also without.

File: ios/Lunes.xcodeproj/project.pbxproj (hunk 3 above)
Comment (SteffiStuffel) [reply ios/Lunes.xcodeproj/project.pbxproj:1058]: Release build is the next issue. Will be adjusted there, if necesarry.

File: .circleci/src/jobs/deliver_browser_stack.yml (hunk 4 above)
Comment (SteffiStuffel) [reply .circleci/src/jobs/deliver_browser_stack.yml:9]: No browserstack credentials are in the environment variables.

File: ios/fastlane/Fastfile (hunk 5 above)
Comment (SteffiStuffel) [reply ios/fastlane/Fastfile:34]: -Creates a new temporary keychain for use with match
-Switches match to readonly mode to not create new profiles/cert on CI
-Sets up log and test result paths to be easily collectible

File: ios/fastlane/Fastfile (hunk 6 above)
Comment (SteffiStuffel) [reply ios/fastlane/Fastfile:36]: I think this is overhead, since it is obvious when looking at the function (second and third line of the function use the parameters), and there is a check if the parameters are missing.

File: .circleci/src/jobs/deliver_browser_stack.yml (hunk 4 above)
Comment (klinzo) [reply .circleci/src/jobs/deliver_browser_stack.yml:9]: Then do we need this here?

Comment (SteffiStuffel) [reply .circleci/src/jobs/deliver_browser_stack.yml]: No, we don't. Removed it.

File: ios/fastlane/Matchfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+# type("appstore") # The default type, can be: appstore, adhoc, enterprise or development
+
+# app_identifier(["tools.fastlane.app", "tools.fastlane.app2"])
+# username("user@fastlane.tools") # Your Apple Developer Portal username
+
+# For all available options run `fastlane match --help`
+# Remove the # in the beginning of the line to enable the other options
+
+# The docs are available on https://docs.fastlane.tools/actions/match
```
Comment (maxammann) [inline ios/fastlane/Matchfile:14]: Minor: You can remove this
```suggestion
```

File: ios/fastlane/Matchfile [hunk 8]
```diff
@@ -0,0 +1,14 @@
+git_url("git@github.com:digitalfabrik/lunes-credentials.git")
+storage_mode("git")
+
+username("delivery@lunes.app")
```
Comment (maxammann) [inline ios/fastlane/Matchfile:4]: From where is this username? Is it a apple account name? If yes, then I think there is no more a need for a special apple account.

We are using now api keys.

**Review by maxammann — APPROVED** [review]

File: ios/fastlane/Matchfile (hunk 8 above)
Comment (SteffiStuffel) [reply ios/fastlane/Matchfile:4]: It is the github-user, which has the ssh key for accessing the credentials repository.
